src/shape.rs

```diff
@@ -48,15 +75,18 @@ pub trait Shape: Sized {
     ///
     /// The `tolerance` parameter is the same as for
     /// [`path_elements()`](#tymethod.path_elements).
+    //TODO: Does this pull its weight? It seems marginal at best.
```

Comment:
Yes, it avoids allocating in the `BezPath` implementation (there, it's free). This was discussed some in #136, see in particular the tables. I think it's worth capturing at least some of that in docs.

Reply:
Does it duplicate the functionality provided by `Shape::as_path_slice`?

Alternatively, we could also have a `as_bezpath` method, similar to `as_rect` or `as_circle`?

Reply:
.. which I guess is what `as_path_slice` is doing, really..

Reply:
It doesn't duplicate that functionality, as that would require allocation to create a BezPath, and this doesn't.

The choice between `as_path_slice` and `as_bezpath` is subtle, and I think boils down to the same reason we generally prefer `&str` in APIs rather than `&String`, though the analogy is not exact; there are some potentially useful methods on `BezPath` that are not implemented on a path slice.